hermes-application/src/test/java/eu/socialedge/hermes/domain/infrastructure/InfrastructureResourceIT.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Hermes - The Municipal Transport Timetable System
+ * Copyright (c) 2016 SocialEdge
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+package eu.socialedge.hermes.domain.infrastructure;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.ClientResponse;
+import com.sun.jersey.api.client.WebResource;
+import com.sun.jersey.api.client.config.DefaultClientConfig;
+import com.sun.jersey.client.urlconnection.URLConnectionClientHandler;
+import eu.socialedge.hermes.application.HermesApplication;
+import eu.socialedge.hermes.application.domain.infrastructure.InfrastructureService;
+import eu.socialedge.hermes.domain.TestDatabaseConfig;
+import eu.socialedge.hermes.domain.TestDatabaseInitializer;
+import eu.socialedge.hermes.domain.geo.dto.LocationSpecification;
+import eu.socialedge.hermes.domain.infrastructure.dto.StationSpecification;
+import eu.socialedge.hermes.domain.infrastructure.dto.StationSpecificationMapper;
+import eu.socialedge.hermes.domain.transport.VehicleType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.MediaType;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+import static eu.socialedge.hermes.domain.TestDatabaseInitializer.*;
+
+@WebIntegrationTest
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {HermesApplication.class, TestDatabaseConfig.class})
+public class InfrastructureResourceIT {
```

Comment:
Maven will not run this one. Only *Test postfix are allowed by default. See http://tomaszdziurko.pl/2013/01/running-unit-tests-integration-tests-separately-maven-testng/